### ISSUE TITLE
File names too long

### ISSUE THREAD

**Issue by coderextreme** [joelittlejohn/jsonschema2pojo#443]
+ ./jsonschema2pojo.sh -T JSON -t X3D/www.web3d.org/x3d/content/examples/Basic -s All2/www.web3d.org/x3d/content/examples/Basic -p ''
Exception in thread "main" java.io.FileNotFoundException: X3D/www.web3d.org/x3d/content/examples/Basic/Geospatial/Appearance_________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________.java (File name too long)
	at java.io.FileOutputStream.open(Native Method)
	at java.io.FileOutputStream.<init>(FileOutputStream.java:213)
	at java.io.FileOutputStream.<init>(FileOutputStream.java:162)
	at com.sun.codemodel.writer.FileCodeWriter.openBinary(FileCodeWriter.java:91)
	at org.jsonschema2pojo.FileCodeWriterWithEncoding.openSource(FileCodeWriterWithEncoding.java:42)
	at com.sun.codemodel.JPackage.createJavaSourceFileWriter(JPackage.java:486)
	at com.sun.codemodel.JPackage.build(JPackage.java:437)
	at com.sun.codemodel.JCodeModel.build(JCodeModel.java:311)
	at org.jsonschema2pojo.Jsonschema2Pojo.generate(Jsonschema2Pojo.java:82)
	at org.jsonschema2pojo.cli.Jsonschema2PojoCLI.main(Jsonschema2PojoCLI.java:48)
+ ./jsonschema2pojo.sh -T JSON -t X3D/www.web3d.org/x3d/content/examples/Vrml2.0Sourcebook -s All2/www.web3d.org/x3d/content/examples/Vrml2.0Sourcebook -p ''
Exception in thread "main" java.io.FileNotFoundException: X3D/www.web3d.org/x3d/content/examples/Vrml2.0Sourcebook/Siggraph98Course/Child______________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________.java (File name too long)
	at java.io.FileOutputStream.open(Native Method)
	at java.io.FileOutputStream.<init>(FileOutputStream.java:213)
	at java.io.FileOutputStream.<init>(FileOutputStream.java:162)
	at com.sun.codemodel.writer.FileCodeWriter.openBinary(FileCodeWriter.java:91)
	at org.jsonschema2pojo.FileCodeWriterWithEncoding.openSource(FileCodeWriterWithEncoding.java:42)
	at com.sun.codemodel.JPackage.createJavaSourceFileWriter(JPackage.java:486)
	at com.sun.codemodel.JPackage.build(JPackage.java:437)
	at com.sun.codemodel.JCodeModel.build(JCodeModel.java:311)
	at org.jsonschema2pojo.Jsonschema2Pojo.generate(Jsonschema2Pojo.java:82)
	at org.jsonschema2pojo.cli.Jsonschema2PojoCLI.main(Jsonschema2PojoCLI.java:48)
+ ./jsonschema2pojo.sh -T JSON -t X3D/www.web3d.org/x3d/content/examples/SavageDefense -s All2/www.web3d.org/x3d/content/examples/SavageDefense -p ''
Exception in thread "main" java.io.FileNotFoundException: X3D/www.web3d.org/x3d/content/examples/SavageDefense/Locations/PearlHarborTerrain/Child______________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________.java (File name too long)
	at java.io.FileOutputStream.open(Native Method)
	at java.io.FileOutputStream.<init>(FileOutputStream.java:213)
	at java.io.FileOutputStream.<init>(FileOutputStream.java:162)
	at com.sun.codemodel.writer.FileCodeWriter.openBinary(FileCodeWriter.java:91)
	at org.jsonschema2pojo.FileCodeWriterWithEncoding.openSource(FileCodeWriterWithEncoding.java:42)
	at com.sun.codemodel.JPackage.createJavaSourceFileWriter(JPackage.java:486)
	at com.sun.codemodel.JPackage.build(JPackage.java:437)
	at com.sun.codemodel.JCodeModel.build(JCodeModel.java:311)
	at org.jsonschema2pojo.Jsonschema2Pojo.generate(Jsonschema2Pojo.java:82)
	at org.jsonschema2pojo.cli.Jsonschema2PojoCLI.main(Jsonschema2PojoCLI.java:48)

**Comment by joelittlejohn** [comment]
It looks like you have so much duplication of names in your input, you have
reached the filename limit as the tool tries to disambiguate the types by
adding an underscore.

I don't think you're heading down a good path here. If you want one type to
represent all these things (instead of duplicate types) then you're going
to have to either use schemas (and refs) or stop feeding in so much
duplicate jason example data (maybe just use one example of each type).

**Comment by coderextreme** [comment]
Well, I’m just trying this out.  Here’s what I did to generate Schema from JSON and Schema (not sure how much of each).

Copy all files in examples to ALL_FOLDER
Remove bad JSON from ALL_FOLDER
Generate Java code from JSON with jsonschema2pojo.sh and put in X3D_FOLDER
Write all Java filenames to JAVA_FILES
Create MERGED_FOLDER
Search for files with pattern [A-Za-z0-9].java in JAVA_FILES and place in MERGED_FOLDER
For each JAVA_FILE in MERGED_FOLDER
	Get the file name without the .java and the folder and place in FILENAME
	For the FILES result of a search for X3D_FOLDER/.../FILENAME_*.java in JAVA_FILES
		Count the words in each file of FILES
		Sort the count reversed, smallest last
		cut the 6th and 7th field
		cut the 2nd field (the 6th or 7th I think)
		Get the first line of the file which doesn have a blank line or 'total'.   Assign to BIGGEST_FILE
		Copy BIGGEST_FILE TO JAVA_FILE
Replace extra _'s in files in MERGED_FOLDER
Replace the package name in the files in MERGED_FOLDER with 'package x3djson;'
Shorten the package name in the variablename and types in the files in MERGED_FOLDER with 'x3djson.SUFFIX'
Generate Java files from XSD with xjc:  xjc -d . -p x3djson x3d-3.4.xsd x3d-3.4-Web3dExtensionsP* -b bindings.xjb
Copy files from xjc generation of files from XSD which aren't in MERGED_FOLDER to the MERGED_FOLDER
Compile the Java files in the MERGED_FOLDER
Make the CLASS_FOLDER
Copy Java classes from MERGED_FOLDER to CLASS_FOLDER
Compile the JSON schema generating class
Send the Java classes in CLASS_FOLDER to the JSON schema generator
Make the SCHEMA_FOLDER
Copy the JSON files generated by the JSON schema generatorinto SCHEMA
Send the Schema to GenSON and retrieve schema.son [ need to verify if the command line got too big ]

I tried this, but it didn’t work:

jsonschema2pojo.sh -T JSONSCHEMA -t schema.json -s SCHEMA_FOLDER


Now I have to validate some JSON with my JSON schema.  LOL. We should probably choose all files generated by xic.
